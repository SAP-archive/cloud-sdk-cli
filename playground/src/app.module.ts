import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';

import { CatalogueModule } from './catalogue/catalogue.module';

@Module({
  imports: [CatalogueModule],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
