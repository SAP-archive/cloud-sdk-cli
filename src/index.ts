/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export { run } from '@oclif/command';
