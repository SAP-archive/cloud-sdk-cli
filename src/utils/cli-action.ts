/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */
import cli from 'cli-ux';

export async function action(output: string, showSpinner: boolean, promise: Promise<any>) {
  if (showSpinner) {
    cli.action.start(output);
  } else {
    cli.log(`${output}...`);
  }
  await promise;
  if (showSpinner) {
    cli.action.stop();
  }
}
