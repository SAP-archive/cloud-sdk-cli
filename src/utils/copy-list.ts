/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface CopyDescriptor {
  sourcePath: string | URL;
  targetFolder: string;
  fileName: string;
}
