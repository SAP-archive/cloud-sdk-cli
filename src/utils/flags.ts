/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

import Command from '@oclif/command';
import { OutputFlags } from '@oclif/parser';

type FlagsInput<CommandT extends typeof Command> = CommandT['flags'];

export type Flags<CommandT extends typeof Command> = OutputFlags<NonNullable<FlagsInput<CommandT>>>;
export type RequiredFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Pick<Flags<CommandT>, ParamsT>;
export type OptionalFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Partial<
  RequiredFlagsParam<CommandT, ParamsT>
>;
export type FlagsParam<
  CommandT extends typeof Command,
  RequiredParamsT extends keyof Flags<CommandT>,
  OptionalParamsT extends keyof Flags<CommandT>
> = RequiredFlagsParam<CommandT, RequiredParamsT> & OptionalFlagsParam<CommandT, OptionalParamsT>;
