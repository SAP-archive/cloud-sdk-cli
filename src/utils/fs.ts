/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */

import { promisify } from 'util';
import * as fs from 'fs';
import * as rimraf from 'rimraf';

export const access = promisify(fs.access);
export const mkdir = promisify(fs.mkdir);
export const readdir = promisify(fs.readdir);
export const readFile = promisify(fs.readFile);
export const copyFile = promisify(fs.copyFile);
export const writeFile = promisify(fs.writeFile);
export const rm = promisify(rimraf);
