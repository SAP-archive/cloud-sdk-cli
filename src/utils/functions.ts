/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface Confirm {
  confirm: (message: string) => Promise<boolean>;
}

function confirm(message: string): Promise<boolean> {
  return Promise.resolve(true);
}

export const testFunctions = {
  confirm
};
