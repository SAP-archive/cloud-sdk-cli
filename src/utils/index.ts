/*!
 * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
 */

export * from './copy-list';
export * from './generate-vdm-util';
export * from './git-ignore';
export * from './jest-config';
export * from './manifest-yaml';
export * from './message-formatter';
export * from './package-json';
export * from './scaffold';
export * from './templates';
export * from './usage-analytics';
export * from './warnings';
