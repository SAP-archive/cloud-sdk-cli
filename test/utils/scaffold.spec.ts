/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */
const confirm = jest.fn().mockResolvedValue(true);
jest.mock('cli-ux', () => {
  // Mocking needs to happen before the command is imported
  const cli = jest.requireActual('cli-ux');
  return {
    ...cli,
    default: {
      ...cli.default,
      confirm
    }
  };
});

import * as fs from 'fs';
import * as path from 'path';
import * as rm from 'rimraf';
import { buildScaffold, shouldBuildScaffold } from '../../src/utils';
import { getCleanProjectDir, getTestOutputDir } from '../test-utils';

const testOutputDir = getTestOutputDir(__filename);

describe('Scaffold Utils', () => {
  afterAll(() => {
    rm.sync(testOutputDir);
  });

  it('should determine if scaffold is needed', async () => {
    const projectDir = getCleanProjectDir(testOutputDir, 'should-build-scaffold');

    expect(await shouldBuildScaffold(projectDir, false)).toBe(true);
    expect(await shouldBuildScaffold(projectDir, true)).toBe(true);

    const packageJsonPath = path.resolve(projectDir, 'package.json');
    fs.copyFileSync(path.resolve('test', 'nest', 'package.json'), packageJsonPath);
    expect(await shouldBuildScaffold(projectDir, false)).toBe(false);
  });

  it('should build the scaffold', async () => {
    const projectDir = getCleanProjectDir(testOutputDir, 'build-scaffold');

    await buildScaffold(projectDir, false, false);

    expect(fs.readdirSync(projectDir).sort()).toMatchSnapshot();
  }, 80000);
});
