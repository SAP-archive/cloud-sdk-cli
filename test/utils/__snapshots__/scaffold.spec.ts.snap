// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scaffold Utils [E2E] should build the scaffold 1`] = `
Array [
  ".git",
  ".gitignore",
  "nest-cli.json",
  "package.json",
  "src",
  "test",
  "tsconfig.build.json",
  "tsconfig.json",
  "tslint.json",
]
`;
